bin/p2-setup-runtime/main.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	dockerclient "github.com/docker/docker/client"
+	"github.com/square/p2/pkg/hooks"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/manifest"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/types"
+	"github.com/square/p2/pkg/uri"
+	"github.com/square/p2/pkg/version"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+const (
+	fullDescription = `Setup runtime environment of an installed pod
+
+EXAMPLES
+
+$ p2-setup-runtime podManifest.yaml
+
+$ p2-setup-runtime https://artifactory.local/podManifest.yaml
+
+$ p2-setup-runtime --pod-root '/var/pods' https://artifactory.local/podManifest.yaml
+`
+)
+
+var (
+	app = kingpin.New("p2-setup-runtime", fullDescription)
+
+	manifestURI = app.Arg("manifest", "a path to a pod manifest").Required().URL()
+	requireFile = app.Flag(
+		"require-file",
+		"If set, the p2-exec invocation(s) written for the pod will not execute until the file exists on the system",
+	).Default("/dev/shm/p2-may-run").String()
+	podRoot       = app.Flag("pod-root", "Pod root directory").Default(pods.DefaultPath).String()
+	hooksDir      = app.Flag("hooks-dir", "Directory hooks run scripts are located").Default(hooks.DefaultPath).String()
+	dockerUrl     = app.Flag("docker-url", "Docker host url").Default(dockerclient.DefaultDockerHost).String()
+	dockerVersion = app.Flag("docker-version", "Docker Version").Default("1.21").String()
+	tlsCA         = app.Flag("tls-ca-file", "File containing the x509 PEM-encoded CA ").ExistingFile()
+	tlsCert       = app.Flag("tls-cert", "File containing tls cert").ExistingFile()
+	tlsKey        = app.Flag("tls-key", "File containing tls key").ExistingFile()
+)
+
+func dockerClient() (*dockerclient.Client, error) {
+	var client *dockerclient.Client
+	var err error
+
+	if *tlsCA != "" && *tlsCert != "" && *tlsKey != "" {
+		client, err = dockerclient.NewClientWithOpts(
+			dockerclient.WithTLSClientConfig(*tlsCA, *tlsCert, *tlsKey),
+			dockerclient.WithHost(*dockerUrl),
+			dockerclient.WithVersion(*dockerVersion),
+		)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		client, err = dockerclient.NewClientWithOpts(
+			dockerclient.WithHost(*dockerUrl),
+			dockerclient.WithVersion(*dockerVersion),
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return client, err
+}
+
+func initPod(manifest manifest.Manifest) (*pods.Pod, error) {
+	nodeName, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+
+	fetcher := uri.DefaultFetcher
+	podReadOnlyPolicy := pods.NewReadOnlyPolicy(false, nil, nil)
+	podFactory := pods.NewFactory(*podRoot, types.NodeName(nodeName), fetcher, *requireFile, podReadOnlyPolicy)
+
+	dockerClient, err := dockerClient()
```

Comment:
I think this marks the THIRD way in our code base that we create the docker client?
https://github.com/square/p2/blob/master/bin/p2-launch/main.go#L97
https://github.com/square/p2/blob/master/pkg/preparer/setup.go#L716

Do we know that `dockerclient.NewClientWithOpts()` is the right choice here vs `NewClient` or `NewEnvClient`? The invocations look very similar on the surface.

Reply:
Earlier versions of initialization methods are now deprecated. so followed suggested way(NewClientWithOpts()) to initialize docker client.